From 8e892529af268a4e86bae045e07c93b5d4c360b0 Mon Sep 17 00:00:00 2001
From: Jan Synacek <jsynacek@redhat.com>
Date: Tue, 27 Nov 2012 12:53:23 +0100
Subject: [PATCH 2/2] arping: Call usage() before limiting capabilities.

Otherwise, running arping binary without the capabilities set results in printing
warnings with the usage.

Signed-off-by: Jan Synacek <jsynacek@redhat.com>
---
 arping.c | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/arping.c b/arping.c
index ccb68a0..02e3d9c 100644
--- a/arping.c
+++ b/arping.c
@@ -959,19 +959,6 @@ main(int argc, char **argv)
 	int socket_errno;
 	int ch;
 
-	limit_capabilities();
-
-#ifdef USE_IDN
-	setlocale(LC_ALL, "");
-#endif
-
-	enable_capability_raw();
-
-	s = socket(PF_PACKET, SOCK_DGRAM, 0);
-	socket_errno = errno;
-
-	disable_capability_raw();
-
 	while ((ch = getopt(argc, argv, "h?bfDUAqc:w:s:I:V")) != EOF) {
 		switch(ch) {
 		case 'b':
@@ -1023,6 +1010,19 @@ main(int argc, char **argv)
 
 	target = *argv;
 
+	limit_capabilities();
+
+#ifdef USE_IDN
+	setlocale(LC_ALL, "");
+#endif
+
+	enable_capability_raw();
+
+	s = socket(PF_PACKET, SOCK_DGRAM, 0);
+	socket_errno = errno;
+
+	disable_capability_raw();
+
 	if (device.name && !*device.name)
 		device.name = NULL;
 
-- 
1.7.11.7

